include/seqan3/alphabet/composite/alphabet_variant.hpp

```diff
@@ -23,109 +24,40 @@
-inline bool constexpr one_alternative_is<alphabet_variant<alternatives...>,
-                                         fun_t,
-                                         target_t> = false;
-
-// guard against tuple composites that contain the variant somewhere (they can implicitly convert at source)
-template <typename ...alternatives,
-          template <typename> typename fun_t,
-          typename target_t>
-    requires alphabet_tuple_base_specialisation<target_t> &&
-             meta::in<detail::transformation_trait_or_t<recursive_tuple_components<target_t>, meta::list<>>,
-                      alphabet_variant<alternatives...>>::value
-inline bool constexpr one_alternative_is<alphabet_variant<alternatives...>,
-                                         fun_t,
-                                         target_t> = false;
-
-// guard against alternatives
-template <typename ...alternatives,
-          template <typename> typename fun_t,
-          typename target_t>
-    requires type_in_pack_v<target_t, alternatives...>
-inline bool constexpr one_alternative_is<alphabet_variant<alternatives...>,
-                                         fun_t,
-                                         target_t> = false;
-
-// guard against alternatives (LHS and RHS switched)
-template <typename ...alternatives,
-          template <typename> typename fun_t,
-          typename target_t>
-    requires type_in_pack_v<target_t, alternatives...>
-inline bool constexpr one_alternative_is<target_t,
-                                         fun_t,
-                                         alphabet_variant<alternatives...>> = false;
-
-// guard against ranges and iterators over self to prevent recursive instantiation
-template <typename ...alternatives,
-          template <typename> typename fun_t,
-          typename target_t>
-    //NO, it's not possible to use the value_type type trait here
-    requires requires { std::same_as<typename target_t::value_type, alphabet_variant<alternatives...>>; }
-inline bool constexpr one_alternative_is<alphabet_variant<alternatives...>,
-                                         fun_t,
-                                         target_t> = false;
-
-// guard against pairs/tuples that *might* contain self to prevent recursive instantiation
-// (applying tuple_like unfortunately does not work because it itself starts recursive instantiation)
-template <typename ...alternatives,
-          template <typename> typename fun_t,
-          typename target_t>
-    requires tuple_size<target_t> && !alphabet_tuple_base_specialisation<target_t>
-inline bool constexpr one_alternative_is<alphabet_variant<alternatives...>,
-                                         fun_t,
-                                         target_t> = false;
-
-//!\endcond
-
+#if SEQAN3_WORKAROUND_GCC_LAZY_REQUIRES
+template <typename t>
+SEQAN3_CONCEPT pseudoalphabet = requires { requires seqan3::alphabet_size<t> > 0; };
```

Comment:
docs?

Reply:
It won't appear in any docs anyway, because the macro is not set by doxygen.

I have renamed it to `variant_guard_pseudoalphabet` that should underline that it is only ever used there.